brush-parser/src/parser.rs

```diff
@@ -587,12 +587,28 @@ peg::parser! {
                 w:word() {
                     ast::CommandPrefixOrSuffixItem::Word(ast::Word::from(w))
                 }
-            )+ { ast::CommandSuffix(s) }
+            )+ e:input_output_pipe_extension()? {
```

Comment:
Would this mean that `|&` is only allowed at the very end of a suffix (and not intermingled with other suffixes)?

Reply:
@reubeno yes, only at the end of the command, where `|` is used. The `command` parser checks if the input has `|&` after all suffixes, and if so, appends the redirection but without consuming `|&`. The `pipe_sequence` parser then splits commands by `|&` or '|'.
https://www.gnu.org/savannah-checkouts/gnu/bash/manual/bash.html#Redirecting-Standard-Output-and-Standard-Error
> If ‘|&’ is used, command1’s standard error, in addition to its standard output, is connected to command2’s standard input through the pipe; it is shorthand for 2>&1 |. <ins>This implicit redirection of the standard error to the standard output is performed after any redirections specified by command1</ins>.

Reply:
It looks like `|&` can be used in the same places that `|` can be used:

![image](https://github.com/user-attachments/assets/83993444-c09b-4e56-915d-c6205e056be8)

...and not just anywhere output redirection (`>`) can be used, e.g.:

![image](https://github.com/user-attachments/assets/ba8cfe90-e593-4afd-9e40-20ce035e1b36)

I think that means the only place where it should really show up in the grammar, at least as it's currently factored, is in `pipe_sequence()`.

Reply:
same in brush. 
```
brush$ ls |&
>
ERROR main: syntax error near token `
' (line 1 col 6)
brush$ |& ls
ERROR main: syntax error near token `|&' (line 1 col 1)
```
`|& ls` is an `io_redirect+` inside cmd_prefix not a `redirect_list`. `redirect_list` is used only at the very end. 
I don't know how to approach `&|` inside the pipe_sequence: `c:command()++ special_operator('|')`. maybe `map` or something?